core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -148,6 +153,12 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         if (executorDataMap.contains(executorId)) {
           executorRef.send(RegisterExecutorFailed("Duplicate executor ID: " + executorId))
           context.reply(true)
+        } else if (scheduler.nodeBlacklist != null &&
+                   scheduler.nodeBlacklist.contains(executorId)) {
+          // Handle a race where the cluster manager finishes creating an executor, just after
+          // the executor is blacklisted and just before it is possibly killed.
+          executorRef.send(RegisterExecutorFailed("Executor is blacklisted: " + executorId))
+          context.reply(true)
```

Comment:
can you change the comment to

If the cluster manager gives us an executor on a blacklisted node (because it already started allocating those resources before we informed it of our blacklist, or if it ignored our blacklist), then we reject that executor immediately.

I'd also add a `logInfo` here about rejecting the executor b/c its on a blacklisted node.